application/server/server.go

```diff
@@ -78,25 +78,26 @@ func Start() {
 const textDocumentDidOpenOperation = "textDocument/didOpen"
 const textDocumentDidSaveOperation = "textDocument/didSave"
 
-func initHandlers(srv *jrpc2.Server, handlers *handler.Map) {
-	(*handlers)["initialize"] = initializeHandler(srv)
-	(*handlers)["initialized"] = initializedHandler(srv)
-	(*handlers)["textDocument/didChange"] = noOpHandler()
-	(*handlers)["textDocument/didClose"] = noOpHandler()
-	(*handlers)[textDocumentDidOpenOperation] = textDocumentDidOpenHandler()
-	(*handlers)[textDocumentDidSaveOperation] = textDocumentDidSaveHandler()
-	(*handlers)["textDocument/hover"] = textDocumentHover()
-	(*handlers)["textDocument/codeAction"] = codeActionHandler()
-	(*handlers)["textDocument/codeLens"] = codeLensHandler()
-	(*handlers)["textDocument/willSave"] = noOpHandler()
-	(*handlers)["textDocument/willSaveWaitUntil"] = noOpHandler()
-	(*handlers)["shutdown"] = shutdown()
-	(*handlers)["exit"] = exit(srv)
-	(*handlers)["workspace/didChangeWorkspaceFolders"] = workspaceDidChangeWorkspaceFoldersHandler(srv)
-	(*handlers)["workspace/willDeleteFiles"] = workspaceWillDeleteFilesHandler()
-	(*handlers)["workspace/didChangeConfiguration"] = workspaceDidChangeConfiguration(srv)
-	(*handlers)["window/workDoneProgress/cancel"] = windowWorkDoneProgressCancelHandler()
-	(*handlers)["workspace/executeCommand"] = executeCommandHandler(srv)
+func initHandlers(srv *jrpc2.Server, handlers handler.Map) {
+	handlers["initialize"] = initializeHandler(srv)
+	handlers["initialized"] = initializedHandler(srv)
+	handlers["textDocument/didChange"] = noOpHandler()
+	handlers["textDocument/didClose"] = noOpHandler()
+	handlers[textDocumentDidOpenOperation] = textDocumentDidOpenHandler()
+	handlers[textDocumentDidSaveOperation] = textDocumentDidSaveHandler()
+	handlers["textDocument/hover"] = textDocumentHover()
+	handlers["textDocument/codeAction"] = handler.New(codeaction.GetCodeActionHandler(di.CodeActionService()))
```

Comment:
It would fit better with the style of the other handlers, to move the `handler.New` into a dedicated method.